cmd/crank/beta/render/runtime_docker.go

```diff
@@ -145,39 +154,80 @@ func GetRuntimeDocker(fn pkgv1beta1.Function, log logging.Logger) (*RuntimeDocke
 	}
 	r := &RuntimeDocker{
 		Image:      fn.Spec.Package,
+		Name:       "",
 		Cleanup:    cleanup,
 		PullPolicy: pullPolicy,
 		log:        log,
 	}
 	if i := fn.GetAnnotations()[AnnotationKeyRuntimeDockerImage]; i != "" {
 		r.Image = i
 	}
+	if i := fn.GetAnnotations()[AnnotationKeyRuntimeNamedContainer]; i != "" {
+		r.Name = i
+	}
 	return r, nil
 }
 
 var _ Runtime = &RuntimeDocker{}
 
-// Start a Function as a Docker container.
-func (r *RuntimeDocker) Start(ctx context.Context) (RuntimeContext, error) {
-	r.log.Debug("Starting Docker container runtime", "image", r.Image)
-	c, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
-	if err != nil {
-		return RuntimeContext{}, errors.Wrap(err, "cannot create Docker client using environment variables")
+func (r *RuntimeDocker) findContainer(ctx context.Context, cli *client.Client, containerName string, startContainer bool) (string, string) {
+	r.log.Debug("searching for Docker container", "name", r.Name)
+
+	filterArgs := filters.NewArgs()
+	filterArgs.Add("name", containerName)
+
+	containers, err := cli.ContainerList(ctx, container.ListOptions{
+		Filters: filterArgs,
+		All:     true, // Include stopped containers
+	})
+
+	if err == nil && len(containers) > 0 {
```

Comment:
I'd prefer to return early here to avoid nesting the bulk of the method's logic inside this conditional.

https://github.com/crossplane/crossplane/tree/master/contributing#return-early

Reply:
done